### PR TITLE
Added Filter UI file 

### LINKED ISSUE #2802: Add Filters in Meshery UI
#### Current Behavior

Meshery server now supports CRUD for WASM filters.

#### Desired Behavior

Meshery should have a Filters page on the UI which will be used work with WASM Filters. 

The UI will look similar to the Patterns UI (/configuration/patterns).

Screenshot from 2021-05-02 10-45-08

Initially it should support CRUD for filter files similar to the current implementation of patterns and later deploy and other functionalities will be added.

One key difference with the implementation of patterns is that the edit option for a filter should only allow the user to specify a different file from a URL or Disk and **NOT** let the users edit the filter file.

#### Implementation

Implementation would be similar to the existing patterns page but including the changes mentioned above.

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

### PR DESCRIPTION
**Pull request by Chadha93**
Signed-off-by: Gaurav Chadha <gauravchadha1676@gmail.com>

**Description**
Added filter UI file for the Meshery UI dashboard (functionalities need to be developed further)
This PR fixes #2802 

**Notes for Reviewers**
Other functionalities still need to be added and integrate it Filter UI.

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Cool!

File: ui/components/Filters.js [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+    if (!sortOrder) sortOrder = "";
+
+    const query = `?page=${page}&page_size=${pageSize}&search=${encodeURIComponent(search)}&order=${encodeURIComponent(
+      sortOrder
+    )}`;
+
+    updateProgress({ showProgress: true });
+
+    dataFetch(
+      `/api/experimental/pattern${query}`,
```
Comment (utkarsh-pro) [inline ui/components/Filters.js:139]: @Chadha93 please change this API endpoint to filter endpoint

File: ui/components/Filters.js [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+      );
+    }
+
+    if (type === "upload") {
+      dataFetch(
+        `/api/experimental/pattern`,
+        {
+          credentials: "include",
+          method: "POST",
+          body: JSON.stringify({ pattern_data: { pattern_file: data }, save: true }),
```
Comment (utkarsh-pro) [inline ui/components/Filters.js:219]: I believe the request body needs to be a different for filters POST API for now // @navendu

File: ui/components/Filters.js [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+  );
+}
+
+function MesheryPatterns({ updateProgress, enqueueSnackbar, closeSnackbar, user, classes }) {
+  const [page, setPage] = useState(0);
+  const [search] = useState("");
+  const [sortOrder] = useState("");
+  const [count, setCount] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const [patterns, setPatterns] = useState([]);
```
Comment (utkarsh-pro) [inline ui/components/Filters.js:108]: You may have to rename this too

File: ui/components/Filters.js [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+          >
+            <DeleteIcon />
+          </IconButton>
+        </Tooltip>
+      </DialogActions>
+    </Dialog>
+  );
+}
+
+function MesheryPatterns({ updateProgress, enqueueSnackbar, closeSnackbar, user, classes }) {
```
Comment (utkarsh-pro) [inline ui/components/Filters.js:102]: Please rename the component name

File: ui/components/Filters.js (hunk 2 above)
Comment (navendu-pottekkat) [reply ui/components/Filters.js:219]: We have not added a POST endpoint yet. And yes, it will be different

File: ui/components/Filters.js (hunk 4 above)
Comment (Chadha93) [reply ui/components/Filters.js]: @utkarsh-pro Thanks for reviewing and providing suggestions. I have pushed the changes below (update-filter). Could you please review it again?

**Comment by leecalcote** [comment]
Ready for review round 2

**Comment by utkarsh-pro** [comment]
This looks great @Chadha93! :+1:

**Review by utkarsh-pro — APPROVED** [review]

**Pull request merged**